osu.Server.DifficultyCalculator/ServerDifficultyCalculator.cs

```diff
@@ -145,49 +146,67 @@ private void processDifficulty(ProcessableBeatmap beatmap, MySqlConnection conn)
                         parameters.ToArray());
                 }
 
-                if (legacyMods == LegacyMods.None && beatmap.Ruleset.RulesetInfo.Equals(beatmap.Beatmap.BeatmapInfo.Ruleset))
+                if (legacyMods == LegacyMods.None && item.Ruleset.RulesetInfo.Equals(item.WorkingBeatmap.BeatmapInfo.Ruleset))
                 {
-                    double beatLength = beatmap.Beatmap.Beatmap.GetMostCommonBeatLength();
+                    double beatLength = item.WorkingBeatmap.Beatmap.GetMostCommonBeatLength();
                     double bpm = beatLength > 0 ? 60000 / beatLength : 0;
 
+                    int countCircle = 0;
+                    int countSlider = 0;
+                    int countSpinner = 0;
+
+                    foreach (var obj in item.WorkingBeatmap.Beatmap.HitObjects.OfType<IHasLegacyHitObjectType>())
+                    {
+                        if ((obj.LegacyType & LegacyHitObjectType.Circle) > 0)
+                            countCircle++;
+                        if ((obj.LegacyType & LegacyHitObjectType.Slider) > 0)
```

Comment:
Passing by as I update https://github.com/ppy/osu-server-beatmap-submission/pull/1 to use `LegacyHitObjectType` too - not sure if it's gonna matter, but BSS included `LegacyHitObjectType.Hold` in this count too, and I'm about to mirror that. Could prove important for mania maps.

Reply:
https://github.com/ppy/osu-difficulty-calculator/pull/233#discussion_r1843568770

Reply:
How... did I miss that D: